IDEAS/LIDEAS/Components/BaseClasses/LinWindowResponse.mo

```diff
@@ -0,0 +1,270 @@
+within IDEAS.LIDEAS.Components.BaseClasses;
+model LinWindowResponse
+  "Shortwave window respones implementation for linearisation"
+  parameter Integer nLay(min=1) "number of layers of the wall";
+  parameter Real[:, nLay + 1] SwAbs
+    "absorbed solar radiation for each layer for look-up table as function of angle of incidence";
+  parameter Real[:, 2] SwTrans
+    "transmitted solar radiation for look-up table as function of angle of incidence";
+  parameter Real[nLay] SwAbsDif
+    "absorbed solar radiation for each layer for look-up table as function of angle of incidence";
+  parameter Real SwTransDif
+    "transmitted solar radiation for look-up table as function of angle of incidence";
+  parameter Boolean linearise = false
+    "Set to true for enabling linearization inputs/outputs";
+  parameter Boolean createOutputs = false
+    "Set to true for enabling linearization inputs/outputs";
+
+  final parameter Integer[nLay] columns=if (nLay == 1) then {2} else integer(
+      linspace(
+      2,
+      nLay + 1,
+      nLay));
+  Modelica.Blocks.Interfaces.RealInput solDir
+    "direct solar illuminance on surface se"
+    annotation (Placement(transformation(extent={{-120,40},{-80,80}})));
+  Modelica.Blocks.Interfaces.RealInput solDif
+    "diffuse solar illuminance on surface s"
+    annotation (Placement(transformation(extent={{-120,0},{-80,40}})));
+  Modelica.Blocks.Interfaces.RealInput angInc "angle of incidence"
+    annotation (Placement(transformation(extent={{-120,-80},{-80,-40}})));
+  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a[nLay] iSolAbs
+    "solar absorptance in the panes"
```

Comment:
technically, it's a heat flow rate, not an absorptance (which I interpret as a coefficient)